### PR TITLE
Vulnerablity Detector - Align efficacy tests with changes in content

### PR DESCRIPTION
**Pull request by sebasfalcone**
|Related issue|
|---|
| #25762  |

## Description
This PR updates the efficacy test to align with the latest changes in the CTI content

### CODE REVIEW

**Comment by sebasfalcone** [comment]
# Changes justification

## 007
- CVE-2023-4736
  - Removed macOS as vulnerable
- CVE-2023-4733
  - Removed macOS as vulnerable	

##  009
- CVE-2009-4129 
  - Added affected range (was affected by default)
- CVE-2007-6715
  - Default status changed to unknown (was affected by default)
- CVE-2008-4059
  - Added affected range (was affected by default)

## 011
- CVE-2010-0653
  - Added affected range (was affected by default)

##  014
- CVE-2023-4736
  - Removed macOS as vulnerable
- CVE-2023-4733
  - Removed macOS as vulnerable	
  
## Policy
Triggered event was not right

File: src/wazuh_modules/vulnerability_scanner/qa/test_data_policy/003/expected_004.out [hunk 1]
```diff
@@ -1,4 +1,4 @@
 [
     "Vulnerability scanner module started",
-    "Event type: 8 processed"
+    "Event type: 7 processed"
```
Comment (Dwordcito) [inline src/wazuh_modules/vulnerability_scanner/qa/test_data_policy/003/expected_004.out:3]: Reason to change this?

Comment (sebasfalcone) [reply src/wazuh_modules/vulnerability_scanner/qa/test_data_policy/003/expected_004.out:3]: Sometimes this test passes it doesn't

This is due to the fact that event queues are not cleaned in between tests

Comment (sebasfalcone) [reply src/wazuh_modules/vulnerability_scanner/qa/test_data_policy/003/expected_004.out:3]: I've added a cleanup for the event queues in between executions

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**